internal/initwd/module_install.go

```diff
@@ -254,6 +270,57 @@ func (i *ModuleInstaller) moduleInstallWalker(ctx context.Context, manifest mods
 					}
 
 					log.Printf("[TRACE] ModuleInstaller: Module installer: %s %s already installed in %s", key, record.Version, record.Dir)
+
+					// Checking for module deprecations in the case no new module versions need installation
+					if addr, isRegistryModule := req.SourceAddr.(addrs.ModuleSourceRegistry); isRegistryModule {
+						regClient := i.reg
+
+						regsrcAddr := regsrc.ModuleFromRegistryPackageAddr(addr.Package)
+						resp, err := regClient.ModuleVersions(ctx, regsrcAddr)
+						if err != nil {
+							log.Printf("[DEBUG] Deprecation for %s could not be checked: call to registry failed: %v", addr.Package.Namespace, err)
+
+						} else {
+						found:
+							for _, modProviderVersions := range resp.Modules {
+								for _, modVersion := range modProviderVersions.Versions {
+									vm, _ := version.NewVersion(modVersion.Version)
+									if vm.Equal(record.Version) {
+										if modVersion.Deprecation != nil {
+											var additionalInfo []string
+											if modVersion.Deprecation.Reason != "" {
+												additionalInfo = append(additionalInfo, modVersion.Deprecation.Reason)
+											}
+											if modVersion.Deprecation.Link != "" {
+												additionalInfo = append(additionalInfo, fmt.Sprintf("More information: %s", modVersion.Deprecation.Link))
+											}
+											var detail string
+											if len(additionalInfo) == 1 {
+												detail = additionalInfo[0]
+											} else {
+												detail = strings.Join(additionalInfo, "\n\n")
+											}
```

Comment:
```suggestion
											var additionalInfo []string
											if modVersion.Deprecation.Reason != "" {
												additionalInfo = append(additionalInfo, modVersion.Deprecation.Reason)
											}
											if modVersion.Deprecation.Link != "" {
												additionalInfo = append(additionalInfo, fmt.Sprintf("More information: %s", modVersion.Deprecation.Link))
											}
											var detail string
											if len(additionalInfo) == 1 {
												detail = additionalInfo[0]
											} else {
												detail = strings.Join(additionalInfo, "\n\n")
											}
```

Could use some suggestion on this bit, as it is repeated elsewhere in `installRegistryModules`. First thought is have this in a separate function, but that seems out of place in this file

Reply:
Sounds good, included 👍 https://github.com/hashicorp/terraform/pull/35231/commits/cacc5dffa2a2aeeb2906c337384bebf8ee0da0e8